src/components/Fieldset/Fieldset.tsx

```diff
@@ -1,70 +1,89 @@
 import clsx from 'clsx';
-import React, { ReactNode } from 'react';
+import React, { ElementType, ReactNode } from 'react';
 import styles from './Fieldset.module.css';
-import Legend from '../Legend';
 
-export interface Props {
+type FieldsetProps = {
   /**
-   * Child node(s) that can be nested inside component
+   * The contents of the fieldset. We suggest a Fieldset.Title followed by
+   * interactive elements.
+   *
+   * Should be wrapped in a fragment to allow our styling to control the spacing
+   * between elements.
    */
-  children?: ReactNode;
+  children: ReactNode;
   /**
-   * CSS class names that can be appended to the component.
+   * Additional classnames passed in for styling.
    */
   className?: string;
+};
+
+type LegendProps = {
   /**
-   * Toggles the visibility of the form control legend
-   */
-  hideLegend?: boolean;
-  /**
-   * HTML legend title text
+   * The legend of the fieldset.
    */
-  legend?: string;
+  children: ReactNode;
   /**
-   * Slot for node to appear directly after legend text. Typically used to include a Toolip
+   * Additional classnames passed in for styling.
    */
-  legendAfter?: ReactNode;
+  className?: string;
+};
+
+type ItemsProps = {
   /**
-   * String for the optional label. By default it is '(optional)'
+   * The content of the control elements in the fieldset.
    */
-  optionalLabel?: string;
+  children: ReactNode;
   /**
-   * Required is passed down to the legend to display "optional" or not
+   * Type of element the immediate wrapper around the contents should be.
+   * @default 'div'
    */
-  required?: boolean;
+  as?: ElementType;
   /**
-   * String for the required label to add additional information if needed.
+   * Additional classnames passed in for styling.
    */
-  requiredLabel?: string;
-}
+  className?: string;
+};
 
 /**
- * Primary UI component for user interaction
+ * Helper sub-component for styling the title at the top of the component.
  */
-export const Fieldset = ({
-  className,
-  legend,
-  legendAfter,
-  hideLegend,
-  required,
+const Legend: React.FC<LegendProps> = ({ children, className }) => {
+  const componentClassName = clsx(className, styles['legend']);
+  return <legend className={componentClassName}>{children}</legend>;
+};
```

Comment:
Code guidelines suggest moving this subcomponent out of this `Fieldset` file into its own file. However, there is an existing `Legend` which seems to be being used outside of the `Fieldset` component and in `<fieldset>` html tags. Should cases be supported? If so should `Fieldset` export its own `Legend`?

Also the existing `Legend` also accepts `required`, `requiredLabel`, and `optionalLabel`. Should we support those?

Reply:
oh interesting ... it looks like only a few use cases, so I'd say easier to consolidate and only use the existing `Legend` component?

I'd be curious to see it used in context, but keeping `required` & `optionalLabel` makes sense to me (to match how we'd handle Inputs) -- less clear when we'd use `requiredLabel`

Reply:
I'm down with a `FieldsetLegend` component but naming wise should we also consider moving/renaming `Fieldset.Items`?
Otherwise we'd end up with something like:
```
import {Fieldset, FieldsetLegend} from '@cz/eds';
<Fieldset>
  <FieldsetLegend>Legend</FieldsetLegend> //or text or something
  <Fieldset.Items>
    {inputs}
  </Fieldset.Items>
</Fieldset>
```